airflow/decorators/base.py

```diff
@@ -296,7 +297,20 @@ class _TaskDecorator(ExpandableFactory, Generic[FParams, FReturn, OperatorSubcla
     @multiple_outputs.default
     def _infer_multiple_outputs(self):
         try:
-            return_type = typing_extensions.get_type_hints(self.function).get("return", Any)
+            # We only care about the return annotation, not anything about the parameters
+            def fake():
+                ...
+
+            fake.__annotations__ = {"return": self.function.__annotations__["return"]}
+
+            return_type = typing_extensions.get_type_hints(fake, self.function.__globals__).get("return", Any)
+        except NameError as e:
+            warnings.warn(
+                "Cannot infer multiple_outputs with forward type references that are not imported. "
```

Comment:
The warning is nice. I think it would be helpful for the message to include the function name too. WDYT?

Something like:
```suggestion
                f"Cannot infer multiple_outputs for TaskFlow function {self.function.__name__!r} with forward type references that are not imported. "
```

Reply:
Oh yes good call.